core/corehttp/hostname.go

```diff
@@ -266,6 +283,19 @@ func isPeerIDNamespace(ns string) bool {
 	}
 }
 
+// Converts an identifier to DNS-safe representation that fits in 63 characters
+func toDNSPrefix(ctx context.Context, api iface.CoreAPI, id string) (prefix string) {
+	// Return as-is if things fit
+	if len(id) <= dnsLabelMaxLength {
+		return id
+	}
+
+	// TODO: get original root
+	// TODO: create a new CID, representing a dns-safe root
```

Comment:
Which APIs would be the most efficient way to create a new unixfs block?
How to ensure its persisted in the datastore?

Reply:
Hm. This is a bit tricky given just the CoreAPI. I guess it would be best to use the `api.Blocks().Put(...)` API for now, but that doesn't work cleanly with native types.